classes/class-wc-rest-connect-migration-flag-controller.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+if ( class_exists( 'WC_REST_Connect_Migration_Flag_Controller' ) ) {
+	return;
+}
+
+class WC_REST_Connect_Migration_Flag_Controller extends WC_REST_Connect_Base_Controller {
+	protected $rest_base = 'connect/migration-flag';
+
+	private function is_valid_state($state) {
+		$valid_states = [
+			WC_Connect_API_Constants::MIGRATION_STATE_NOT_STARTED,
+			WC_Connect_API_Constants::MIGRATION_STATE_STARTED,
+			WC_Connect_API_Constants::MIGRATION_STATE_COMPLETED,
+			WC_Connect_API_Constants::MIGRATION_STATE_FAILED
+		];
+		if (in_array($state, $valid_states, true)) {
```

Comment:
I had this as `$state >= 1 && $state <= 4` because it's simpler and more efficient than `in_array`, but that smelled a bit, especially if we started removing constants and skipping integer sequences. I thought about adding an enum class but that's too complex for this.

At the end, I used a simple array of constants just so that we could verify the input.

Reply:
I like the WC_Connect_WCST_To_WCShipping_Migration_State_Enum suggestion below. I will update it with an enum class.

Reply:
Updated in 85a7e6148804edfe858e56f13d0e1fe94ad05611